src/util-thash.c

```diff
@@ -516,7 +581,10 @@ static THashData *THashDataGetNew(THashTableContext *ctx, void *data)
 
     // setup the data
     BUG_ON(ctx->config.DataSet(h->data, data) != 0);
-
+    if (ctx->config.DataSize) {
+        uint32_t len = ctx->config.DataSize(data);
+        (void)SC_ATOMIC_ADD(ctx->memuse, (uint64_t)len);
```

Comment:
where do we actually check if the data we intend to store fits the memcap?

Reply:
Every time a dataset is added, a call is made to `THashGetFromHash` which returns either the found `THashData` or creates new data using `THashDataGetNew`  <-- this checks for `memcap`.

After a dataset has been added, there's a call made to `THashConsolidateMemcap` <- this fn tends to update the `memcap` as per the current `memuse`. This fn is the reason, I chose to increment `memuse` as the rest should be taken care of as it is for datasets of the other types as well.
wdyt?

Reply:
But we shouldn't add data if it violates the memcap, and it seems we're not checking.

Reply:
indeed. The memuse update is happening post addition. Not sure why did I not see it. 😓